addons/sale_project/models/project.py

```diff
@@ -36,12 +36,26 @@ def _compute_sale_line_id(self):
     @api.depends('sale_order_id.invoice_status', 'tasks.sale_order_id.invoice_status')
     def _compute_has_any_so_to_invoice(self):
         """Has any Sale Order whose invoice_status is set as To Invoice"""
-        project_to_invoice = self.env['project.project'].search([
-            '|',
-                ('sale_order_id.invoice_status', '=', 'to invoice'),
-                ('tasks.sale_order_id.invoice_status', '=', 'to invoice'),
-            ('id', 'in', self.ids)
-        ])
+        if not len(self.ids):
```

Comment:
What about this?
```suggestion
        if not self:
```

Reply:
The problem I had came when the record is not created yet and that ids is empty as it only contains a new_id.
This is a test that crashed.
So I think self still exists, its only ids that is empty which leads to an incorrect query, isn't it?

Reply:
I'll give it a try... Thanks 🙂

Reply:
Oh, if it is about new_ids, then you should probably call `self.flush()`, but I'm not sure about the implications to call it in a compute method :fearful:

Reply:
Never work too late... I did forget the return 🙈